[Security] [LoginLink] Add technical details (#17872)
symfony/symfony-docs: security/login_link.rst

    properties. Whenever these change, the hash changes and previous login
    links are invalidated.

For a user that returns ``user@example.com`` on ``$user->getUserIdentifier()``
call, the generated login link looks like this:

> .. code-block:: text

Reviewer: ```suggestion
.. code-block:: text

```

    properties. Whenever these change, the hash changes and previous login
    links are invalidated.

For a user that returns ``user@example.com`` on ``$user->getUserIdentifier()``

> call, the generated login link looks like this: ``http://example.com/login_check?user=user@example.com&expires=1675707377&hash=f0Jbda56Y...A5sUCI~TQF701fwJ...7m2n4A~``

Reviewer: ```suggestion
call, the generated login link looks like this:

.. code-block:: text
    http://example.com/login_check?user=user@example.com&expires=1675707377&hash=f0Jbda56Y...A5sUCI~TQF701fwJ...7m2n4A~
```

?

        </form>
    {% endblock %}

> Hashing strategy

Reviewer: ```suggestion
Hashing Strategy
```

~~~~~~~~~~~~~~~~

Internally, the :class:`Symfony\\Component\\Security\\Http\\LoginLink\\LoginLinkHandler`
implementation uses the
:class:`Symfony\\Component\\Security\\Core\\Signature\\SignatureHasher` to create the
hash contained in the login link.

This hasher creates a first hash with the expiration
date of the link, the values of the configured signature properties and the
user identifier. The used hashing algorithm is SHA-256.

> Once this first hash is processed and encoded in base 64, a new one is created

Reviewer: ```suggestion
Once this first hash is processed and encoded in Base64, a new one is created
```

from the first hash value and the ``kernel.secret`` container parameter. This
allows Symfony to sign this final hash, which is contained in the login URL.

> This final hash is also a base 64 encoded SHA-256 hash.

Reviewer: ```suggestion
The final hash is also a Base64 encoded SHA-256 hash.
```
